libcontainer/specconv/spec_linux.go

```diff
@@ -650,13 +654,29 @@ func stringToDeviceRune(s string) (configs.DeviceType, error) {
 	}
 }
 
-func createDevices(spec *specs.Spec, config *configs.Config) error {
-	// Add default set of devices.
-	for _, device := range AllowedDevices {
-		if device.Path != "" {
-			config.Devices = append(config.Devices, device)
+func createDevices(spec *specs.Spec, config *configs.Config) ([]*configs.Device, error) {
+
+	// If a spec device is redundant with a default device, remove that default
+	// device (the spec one takes priority).
+	dedupedAllowDevs := []*configs.Device{}
+
+	if spec.Linux != nil {
```

Comment:
Hmm, I think you've changed the logic of this function in case `spec.Linux == nil`. Previously it did something (added entries to config.Devices), now it does not.

Reply:
Hi @kolyshkin, thanks for the review. Good catch here, I've re-worked the code.